lib/pure/os.nim

```diff
@@ -3514,3 +3579,7 @@ when not defined(nimscript):
     template existsDir*(args: varargs[untyped]): untyped {.deprecated: "use dirExists".} =
       dirExists(args)
   # {.deprecated: [existsFile: fileExists].} # pending bug #14819; this would avoid above mentioned issue
+
+func isValidFilename*(path: string, maxLen = windowsFilenameMaxLen): bool {.since: (1, 1), deprecated: "use `path.extractFilename.isPortableFilename`".} =
+  ## Returns `path.extractFilename.isPortableFilename(maxLen)`.
+  result = path.extractFilename.isPortableFilename(maxLen)
```

Comment:
Since you changed the implementation to the suggested workaround, there is no need to deprecate it at all.

Reply:
There is still the fact that this does not (and arguably cannot) guarantee that a filename is valid.

@Araq Seriously, how many different situations are there that require something like this? The only programming language library I've seen (out of 5 languages*) that has this functionality is Boost, which is practically the embodiment of "the kitchen sink".

*Go, Python, Ruby, Java, C++

Reply:
Haven't you noticed? "We need X, the other languages also offer it" is always a valid argument, "We don't need X, the other langs don't offer it either" isn't one. Yes, I don't really understand this either.

Reply:
Originally I was just try to make it more safe, so users can not crash your app y using filenames with "lpt" or "con".

I think that checking whole paths is not possible, basically because you can mount anything as a drive on Linux and BSD.
Checking if a filename is usable on other OS is doable and cheap.
Maybe deprecate both proc and leave the all the `const` and let user make their own check.

About guarantees, remember that [Nim can not differentiate between a file and a folder](https://github.com/nim-lang/Nim/issues/16388) yet, so theres a lot that can not be 100% sure.

Reply:
> About guarantees, remember that Nim can not differentiate between a file and a folder yet,
so theres a lot that can not be 100% sure.

Yawn...